### PR TITLE
Feature/fres 1590 story quizz no buttons

### PR DESCRIPTION
**Pull request by maxired**
this PR removes buttons on the start and end screen and instead use the vote mechanism already implement for the game

Here what the three screen looks like

Screenshot 2022-08-22 at 12 01 16
Screenshot 2022-08-22 at 12 01 24
Screenshot 2022-08-22 at 12 01 34

### CODE REVIEW

File: elements/reigns/src/Meter.tsx [hunk 1]
```diff
@@ -38,10 +50,14 @@ export const Meter = ({
...
-  }, [percent]);
+  }, [percent, phase]);
 
   return (
-    <div className={"meter"}>
+    <div className={clsx("meter",
+        phase === GamePhase.NOT_STARTED && 'meter--game-not-started',
+        phase === GamePhase.ENDED && 'meter--game-ended',
+        phase === GamePhase.STARTED && 'meter--game-started',
+      )}>
```
Comment (staff0rd) [inline elements/reigns/src/Meter.tsx:60]: `clsx` also supports:

```suggestion
    <div className={clsx("meter", {
      "meter--game-not-started": phase === GamePhase.NOT_STARTED,
      "meter--game-ended": phase === GamePhase.ENDED,
      "meter--game-started": phase === GamePhase.STARTED,
    })}>
```

File: elements/reigns/src/features/voting/useVoteListener.ts [hunk 2]
```diff
@@ -7,10 +7,20 @@ import { persistParticipantVote } from "./persistence";
 
 export const useVoteListener = (phase: GamePhase) => {
   useEffect(() => {
-    if (phase === GamePhase.STARTED) {
+    if (
+      phase === GamePhase.STARTED ||
+      phase === GamePhase.NOT_STARTED ||
+      phase === GamePhase.ENDED
```
Comment (staff0rd) [inline elements/reigns/src/features/voting/useVoteListener.ts:13]: this is every `GamePhase` member, do we still need this `if`?

File: elements/reigns/src/screens/NotStartedScreen.tsx [hunk 3]
```diff
@@ -1,32 +1,41 @@
...
-  doRestartGame,
 }: {
   gameDefinition: GameDefinition | null;
   isHost: string | boolean | undefined;
-  doRestartGame: () => void;
 }) {
-  const messageRef = useTextFit(gameDefinition?.gameName);
+  const votes = useVotes();
+  if (!gameDefinition) {
+    // todo loading
```
Comment (staff0rd) [inline elements/reigns/src/screens/NotStartedScreen.tsx:17]: todo

File: elements/reigns/src/screens/VotingScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        />
+
+        <div className="answer">
+          <div className="round">
+            {round !== 0 && `${gameDefinition.roundName} ${round}`}
+          </div>
+        </div>
+        <OptionalAnswerText
+          visible={Boolean(selectedCard.answer_yes)}
+          text={selectedCard.answer_yes || "Yes"}
```
Comment (staff0rd) [inline elements/reigns/src/screens/VotingScreen.tsx:65]: Maybe these `|| "Yes"` and `|| "No"` above don't make sense anymore considering we do support empty `text`.

File: elements/reigns/src/features/game/Game.ts [hunk 5]
```diff
@@ -42,6 +48,20 @@ export class Game {
     this.clearVotes();
   }
 
+  prepareGame(definition: GameDefinition) {
+    this.persist({
+      phase: GamePhase.NOT_STARTED,
+      selectedCard: {
+        card: definition.gameName,
+        answer_yes: "Move on the purple area to start the game",
+      } as Card,
```
Comment (staff0rd) [inline elements/reigns/src/features/game/Game.ts:57]: This type assertion, also used in `EndedScreen` indicates `type PersistedGameState { ... selectedCard: Card | null` is not accurate. The `selectedCard` union type could be extended to better communicate this.

**Review by staff0rd — APPROVED** [review]

**Comment by maxired** [comment]
One thing I realized with this PR is that the card fill animation is not working on start and end screen.
will not solve in this PR, has it might require a refactor

File: elements/reigns/src/features/game/Game.ts [hunk 6]
```diff
@@ -52,8 +52,10 @@ export class Game {
     this.persist({
       phase: GamePhase.NOT_STARTED,
       selectedCard: {
+        id: GamePhase.NOT_STARTED,
         card: definition.gameName,
-        answer_yes: "Move on the purple area to start the game",
+        answer_yes: "Move on the purple area the game",
```
Comment (staff0rd) [inline elements/reigns/src/features/game/Game.ts:57]: This change is odd - the new sentence doesn't make sense.

File: elements/reigns/src/features/game/types.ts [hunk 7]
```diff
@@ -39,7 +39,14 @@ export type Card = {
   conditions: string;
 };
 
-export type MenuCard = Pick<Card, "card" | "answer_yes">;
+// a Card that has no impact, weight or stats, that is used by the game as menu screen (start, restart)
+export type MenuCard = Pick<Card, "card" | "answer_yes" | "answer_no"> & {
+  id: GamePhase.ENDED | GamePhase.NOT_STARTED;
+};
+
+export function isMenuCard(card: Card | MenuCard): card is MenuCard {
```
Comment (staff0rd) [inline elements/reigns/src/features/game/types.ts:47]: nice, I don't use user-defined type guards enough.

**Review by staff0rd — APPROVED** [review]

**Pull request merged**